src/dmd/dmodule.d

```diff
@@ -231,6 +232,17 @@ extern (C++) class Package : ScopeDsymbol
         return "package";
     }
 
+    override bool equals(const RootObject o) const
+    {
+        // custom 'equals' for bug 17441. "package a" and "module a" are not equal
+        if (this == o)
+            return true;
+        auto p = cast(Package)o;
+        if (p && isModule() == p.isModule() && ident.equals(p.ident))
+            return true;
+        return false;
+    }
```

Comment:
```d
return p && isModule() == p.isModule && ident.equals(p.ident);
```

Reply:
OK, but you dropped the parens in `p.isModule`. Is it on purpose?

Reply:
It complies either way, but I'd leave it there.

Reply:
> OK, but you dropped the parens in `p.isModule`. Is it on purpose?

No, I should have copied and pasted instead of typing it out. Sorry about that.